addons/binding/org.openhab.binding.freebox/src/main/java/org/openhab/binding/freebox/internal/api/FreeboxApiManager.java

```diff
@@ -0,0 +1,572 @@
+
+            if (patchTableReponse) {
+                // Replace empty result by an empty table result
+                jsonResponse = jsonResponse.replace("\"result\":{}", "\"result\":[]");
+            }
+
+            return evaluateJsonReesponse(jsonResponse, responseClass, doNotLogData);
+        } catch (FreeboxException e) {
+            if (retryAuth && e.isAuthRequired()) {
+                logger.debug("Authentication required: open a new session and retry the request");
+                openSession();
+                return executeUrl(httpMethod, relativeUrl, requestContent, responseClass, false, patchTableReponse,
+                        doNotLogData);
+            }
+            throw e;
+        } catch (IOException e) {
+            throw new FreeboxException(
+                    httpMethod + " request " + relativeUrl + ": execution failed: " + e.getMessage());
+        } catch (JsonSyntaxException e) {
+            throw new FreeboxException(
+                    httpMethod + " request " + relativeUrl + ": response parsing failed: " + e.getMessage());
+        }
+    }
+
+    private <T extends FreeboxResponse<F>, F> F evaluateJsonReesponse(String jsonResponse, Class<T> responseClass,
+            boolean doNotLogData) throws JsonSyntaxException, FreeboxException {
+        logger.debug("evaluateJsonReesponse Json {}", doNotLogData ? "***" : jsonResponse);
+        // First check only if the result is successful
+        FreeboxResponse<Object> partialResponse = gson.fromJson(jsonResponse, EmptyResponse.class);
+        partialResponse.evaluate();
+        // Parse the full response in case of success
+        T fullResponse = gson.fromJson(jsonResponse, responseClass);
+        fullResponse.evaluate();
+        F result = fullResponse.getResult();
+        return result;
+    }
+
+    private String encodeUrl(String url) throws FreeboxException {
+        try {
+            return URLEncoder.encode(url, "UTF-8");
+        } catch (UnsupportedEncodingException e) {
+            throw new FreeboxException("Encoding the URL \"" + url + "\" in UTF-8 failed");
+        }
+    }
+
+    private static String hmacSha1(String key, String value) throws FreeboxException {
+        try {
+            // Get an hmac_sha1 key from the raw key bytes
+            byte[] keyBytes = key.getBytes();
+            SecretKeySpec signingKey = new SecretKeySpec(keyBytes, "HmacSHA1");
+
+            // Get an hmac_sha1 Mac instance and initialize with the signing key
+            Mac mac = Mac.getInstance("HmacSHA1");
+            mac.init(signingKey);
+
+            // Compute the hmac on input data bytes
+            byte[] rawHmac = mac.doFinal(value.getBytes());
+
+            // Convert raw bytes to Hex
+            byte[] hexBytes = new Hex().encode(rawHmac);
+
+            // Covert array of Hex bytes to a String
+            return new String(hexBytes, "UTF-8");
+        } catch (Exception e) {
```

Comment:
It feels like something where only some predefined exceptions will be given, I would prefer if you catch those.

Reply:
Done